File: CarRacer/app/src/main/java/com/carracer/infrastructure/network/models/Responses/LeaderboardResponse.java
package com.carracer.infrastructure.network.models.Responses;

import com.carracer.infrastructure.network.models.Dtos.LeaderboardDto;
import com.google.gson.annotations.SerializedName;
import java.util.List;

public class LeaderboardResponse {
    @SerializedName("isSuccess")
    private boolean isSuccess;
    @SerializedName("message")
    private String message;
    @SerializedName("leaderboard")
    private List<LeaderboardDto> leaderboard;

    public boolean isSuccess() { return isSuccess; }
    public String getMessage() { return message; }
    public List<LeaderboardDto> getLeaderboard() { return leaderboard; }
}

File: CarRacerAPI/Responeses/LeaderboardResponse.cs
using CarRacerAPI.Dtos;

namespace CarRacerAPI.Responeses;

public class LeaderboardResponse
{
    public bool IsSuccess { get; set; }
    public string? Message { get; set; }
    public List<LeaderboardDto> Leaderboard { get; set; } = new List<LeaderboardDto>();
}